articles/javascript-iterations-which-one-is-faster/index.md

```diff
@@ -0,0 +1,755 @@
+const arraySize = new Array(iterations)
+
+console.log("Starting speed/execution test performance with  arraySize iterations" , iterations);
+console.log("arraySize", + arraySize.length)
+
+//for
+console.time("for")
+for (let i = 0; i <iterations; i++) {}
+console.timeEnd("for");
+
+//while
+console.time("while")
+let i = 0;
+while (i < iterations) {
+  i++;
+}
+console.timeEnd("while");
+
+//do...while
+console.time("do...while")
+let ie = 0;
+do {ie++;}
+while (ie < iterations);
+console.timeEnd("do...while");
+
+//for...in
+console.time("for...in")
+for (let ele in arraySize){}
+console.timeEnd("for...in");
+
+//for...of
+console.time("for...of")
+for (let ele of arraySize){}
+console.timeEnd("for...of");
+
+//for...each
+console.time("for...each")
+arraySize.forEach(value=> "")
+console.timeEnd("for...each");
+```
+
+##### Results
+
+```js
+arraySize iterations 100
+for: 0.968ms
+while: 0.047ms
+do...while: 0.045ms
+for...of: 0.113ms
+for...each: 0.065ms
+```
+
+The results are quite eye-catching with `while`, `do-while`, and `for` comparatively faster. To get a clear picture and more desirable result to try to increase the number of iterations.
+
+```js
+arraySize iterations 1000
+for: 0.519ms
+while: 0.129ms
+do...while: 0.215ms
+for...of: 1.012ms
+for...each: 0.103ms
+
+arraySize iterations 10000
+for: 1.038ms
+while: 1.418ms
+do...while: 1.784ms
+for...of: 4.314ms
+for...each: 0.323ms
+
+arraySize iterations 100000
+for: 17.362ms
+while: 8.722ms
+do...while: 9.565ms
+for...of: 29.333ms
+for...each: 1.160ms
+
+arraySize iterations 1000000
+for: 17.613ms
+while: 13.025ms
+do...while: 13.498ms
+for...of: 96.863ms
+for...each: 12.207ms
+
+arraySize iterations 10000000
+for: 51.083ms
+while: 42.198ms
+do...while: 46.366ms
+for...of: 762.958ms
+for...each: 115.888ms
+
+arraySize iterations 100000000
+for: 358.317ms
+while: 344.164ms
+do...while: 349.297ms
+for...of: 14124.834ms
+for...each: 11820.696ms
+```
+
+As the number of iterations increases, the results become quite comparable, and again the old `for` loop is giving stiff competition to the `while` and `do-while` loops. On the other hand, `for` of loop seems to lag behind.
+
+##### Nested Loop Performance
+```js
+const iterations = 100000
+// add an empty array size equal to the number of iterations
+const arraySize = new Array(iterations)
+console.log("arraySize", + arraySize.length)
+
+//nested loop
+console.time("nested loop")
+for (let i = 0; i < iterations; i++) {
+   for (let j = 0; j < iterations; j++) {
+   }
+}
+console.timeEnd("nested loop");
+```
+
+Created a separate code block because the nested loop creates performance overhead as the number of iterations increases.
+
+Test the loop up to 100000 iterations.
+
+##### Results
+
+```js
+arraySize 100: 3.554ms
+arraySize 1000: 71.934ms
+arraySize 10000: 490.257ms
+arraySize 100000: 36745.962ms
+arraySize 100000: 38987.654ms
+```
+
+#### Using Performance Observer perf_hooks
```

Comment:
This leads to some sort of infinite loop. It runs for some minutes before stopping.

Reply:
Yeah, calculates the average time, I will add a note paragraph for that @LinusMuema

Reply:
Any feedback on this issue @kimkimani ?

Reply:
added a note @LinusMuema